src/audiovideocontroller/AudioVideoControllerState.ts

```diff
@@ -112,6 +112,17 @@ export default class AudioVideoControllerState {
   // that we include in the `videosToReceive`.
   videoSubscriptionLimit: number = 25;
 
+  // The previous SDP offer will be used as a dictionary to seed the compression library
+  // during sending of the sdp offer in the Subscribe Frame.
+  previousSdpOfferAsString: string = '';
```

Comment:
Oops, you can just use `previousSdpOffer.sdp`, as `sdp` is public.

Reply:
I am not able to access previousSdpOffer.sdp, I guess because the previousSdpOffer is of type SDP, which is an  interface and it only exposes functions.

If I was using DefaultSDP then I would have been able to access it as its public in DefaultSDP class.

Reply:
ergh thats so silly. maybe in 4.0 we will clean up those headaches.

Reply:
tbh often we just try to downcast to the default version. but this is fine.

Reply:
Actually this is v3 can we just remove that interface and use a SDP class?

Reply:
For v2, I think we can cast it to DefaultSDP